plugins/package-managers/gleam/src/main/kotlin/GleamDependencyHandler.kt

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (C) 2025 The ORT Project Copyright Holders <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.gleam
+
+import java.io.File
+
+import org.ossreviewtoolkit.downloader.VcsHost
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.HashAlgorithm
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.SourceCodeOrigin
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.utils.DependencyHandler
+import org.ossreviewtoolkit.model.utils.prependPath
+import org.ossreviewtoolkit.utils.ort.normalizeVcsUrl
+
+import org.semver4j.Semver
+import org.semver4j.range.RangeListFactory
+
+/**
+ * Context containing all project-level information needed for dependency resolution.
+ */
+internal data class GleamProjectContext(
+    val hexClient: HexApiClient,
+    val project: Project,
+    val workingDir: File,
+    val gleamToml: GleamToml,
+    val manifest: GleamManifest
+)
+
+/**
+ * A sealed interface representing a Gleam dependency that can be either:
+ * - From the manifest.toml lockfile (already resolved with transitive deps)
+ * - From gleam.toml (needs dynamic resolution, no transitive deps)
+ */
+internal sealed interface GleamPackageInfo {
```

Comment:
This interface in total is quite large. Please consider moving it to a separate `GleamPackageInfo.kt` file, and also to move the `data class` implementation to the top level in that same file for a better overview. (Similar to like in `scanner/src/main/kotlin/ScannerWrapper.kt`.)

Reply:
do you mind me doing this in a separate refactor commit at the end? splitting this back into the others is a bit of a nuisance.